File: src/main/java/de/huddeldaddel/euler/Problem10.java
package de.huddeldaddel.euler;

import de.huddeldaddel.euler.math.PrimeSequence;
import de.huddeldaddel.euler.math.PrimeSequence;

public class Problem10 {

    public static void main(String[] args) {
        System.out.println(new Problem10().getResult());
    }

    public long getResult() {
        long result = 0;
        long prime;
        final PrimeSequence primeSequence = new PrimeSequence();
        while((prime = primeSequence.nextValue()) < 2_000_000)
            result += prime;
        return result;
    }

}

File: src/main/kotlin/de/huddeldaddel/euler/Problem10.kt
package de.huddeldaddel.euler

import de.huddeldaddel.euler.math.PrimeSequence

/**
 * Solution for https://projecteuler.net/problem=10
 */
fun main() {
    println(Problem10().result)
}

class Problem10 {

    val result: Long
        get() {
            var result: Long = 0
            var prime: Long
            val primeSequence = PrimeSequence()
            prime = primeSequence.nextValue()
            while (prime < 2000000) {
                result += prime
                prime = primeSequence.nextValue()
            }
            return result
        }

}
